### ISSUE TITLE
nvme verbose log printed with "nvme list" cmd

### ISSUE THREAD

**Issue by yizhanglinux** [linux-nvme/nvme-cli#1593]
Hello
I compiled the latest nvme-cli, found some nvme verbose log output even with "nvme list" cmd, is that expected?

```
$ uname -r
5.19.0-rc5
$ nvme --version
nvme version 2.0 (git 2.0-165-gc9b7d00)
libnvme version 1.0 (git 1.0-105-ge757ffe)

$ nvme list
Node                  Generic               SN                   Model                                    Namespace Usage                      Format           FW Rev  
--------------------- --------------------- -------------------- ---------------------------------------- --------- -------------------------- ---------------- --------
/dev/nvme0n1          /dev/ng0n1            S48CNC0N400972B      Samsung SSD 983 DCT 960GB                1           0.00   B / 960.20  GB    512   B +  0 B   EDA5302Q

$ dmesg
[ 2408.204381] nvme0: Identify(0x6), Invalid Field in Command (sct 0x0 / sc 0x2) MORE

$ nvme show-regs /dev/nvme0n1
cap     : 3028033fff
version : 10200
cc      : 460001
csts    : 1
nssr    : 0
crto    : 0
intms   : 0
intmc   : 0
aqa     : 1f001f
asq     : 284baf000
acq     : 284bae000
cmbloc  : 0
cmbsz   : 0
bpinfo  : 0
bprsel  : 0
bpmbl   : 0
cmbmsc	: 0
cmbsts	: 0
pmrcap  : 0
pmrctl  : 0
pmrsts  : 0
pmrebs  : 0
pmrswtp : 0
pmrmscl	: 0
pmrmscu	: 0

$ dmesg
[ 2578.505337] nvme0: Identify(0x6), Invalid Field in Command (sct 0x0 / sc 0x2) MORE 
[ 2578.518015] nvme0: Unknown(0x7f), Invalid Command Opcode (sct 0x0 / sc 0x1) 
[ 2578.528673] nvme0: Identify(0x6), Invalid Field in Command (sct 0x0 / sc 0x2) MORE 

```

**Comment by yizhanglinux** [comment]
> The messages are coming from the kernel, and they are expected. We may be able to avoid some of them, though.

Thanks for the update, so is it something we need to improve from the kernel or nvme-cli side?

**Comment by keithbusch** [comment]
I think the kernel is purposefully logging these. Perhaps it's too noisy and we can discuss on that list about suppressing them since they are a bit alarming.

On the tooling side, we might know to avoid certain commands if we know they'll fail. There are some that we have no way to know if they're valid without trying it though, so those would always get an error that the current kernel will log.

**Comment by tbzatek** [comment]
Related: https://bugzilla.kernel.org/show_bug.cgi?id=215763

Has there been any follow-up discussion on the kernel side?

**Comment by igaw** [comment]
I am not aware of any discussion on the nvme mailing list, but don't take my word on it. I might just missed it. But please take this to the mailing list.

**Comment by igaw** [comment]
https://lore.kernel.org/linux-nvme/20220929171338.10792-1-dwagner@suse.de/

**Comment by keithbusch** [comment]
Well, the verbose logging is already suppressed for internally generated admin commands. If we suppress it for user commands as well, there's nothing left, so the code would be dead.

**Comment by igaw** [comment]
So a more fine grained approach? Just filtering out the Identify(0x6) command?

**Comment by keithbusch** [comment]
I think we'd have to reach out to the developer who introduced the verbose logging to find out if these are the kinds of errors are what they needed to capture (I personally think it's too noisy).

I think ideally it would have been nice if the nvme ioctl struct had some 'flags' field and we could have defined one that could have set RQF_QUIET in the rq_flags.

**Comment by igaw** [comment]
Okay, thanks for the tips. I'll look into this.

**Comment by keithbusch** [comment]
The kernel side to quiet these is merged for 6.1; let me see if stable-bot picks it up for the older releases.
